README.md

```diff
@@ -20,62 +91,429 @@ Add the following to your `pubspec.yaml` file:
+        return TextSpan(text: "🐦");
       },
-      onImageTap: (src) {
-        // Display the image in large form.
+      "flutter": (RenderContext context, Widget child, Map<String, String> attributes, _) {
+        return FlutterLogo(
+          style: (attributes['horizontal'] != null)
+              ? FlutterLogoStyle.horizontal
+              : FlutterLogoStyle.markOnly,
+          textColor: context.style.color,
+          size: context.style.fontSize.size * 5,
+        );
       },
+    },
+);
+```
+</details>
+
+2. Complex example - rendering an `iframe` differently based on whether it is an embedded youtube video or some other embedded content
+
+Packages used: [`data_connection_checker`](https://pub.dev/packages/data_connection_checker) and [`flutter_inappwebview`](https://pub.dev/packages/flutter_inappwebview)
+
+<details><summary>View code</summary>
+
+```dart
+class CustomRenderExample extends StatefulWidget {
+  CustomRenderExample({Key key, this.title}) : super(key: key);
+  
+    final String title;
+  
+    @override
+    CustomRenderExampleState createState() => new CustomRenderExampleState();
+}
+
+class CustomRenderExampleState extends State<CustomRenderExample> {
+  bool result = false;
+  
+  @override
+  void initState() async {
+    // store current internet connection status
+    result = await DataConnectionChecker().hasConnection;
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: AppBar(
+        title: Text('flutter_html Example'),
+        centerTitle: true,
+      ),
+      body: SingleChildScrollView(
+        child: Html(
+             data: """
+             <h3>YouTube iframe:</h3>
+             <iframe src="https://google.com"></iframe>
+             <h3>Google iframe:</h3>
+             <iframe src="https://www.youtube.com/embed/tgbNymZ7vqY"></iframe>
+             """,
+             customRender: {
+                 "iframe": (RenderContext context, Widget child, Map<String, String> attributes, _) {
+                   // check internet connection
+                   if (!result) {
+                     return Container(
+                       child: ListTile(
+                         leading: Icon(Icons.error, color: Colors.red),
+                         title: Text("Cannot load iframe. You are offline, please check connection and try again.")
+                       )
+                     );
+                   } else if (attributes != null) {
+                     double width = double.tryParse(attributes['width'] ?? "");
+                     double height = double.tryParse(attributes['height'] ?? "");
+                     print(attributes['src']);
+                     return Container(
+                       width: width ?? (height ?? 150) * 2,
+                       height: height ?? (width ?? 300) / 2,
+                       child: InAppWebView(
```

Comment:
Isn't this a webview implementation from some extra dependency? As in, not flutter_webview, right?

Reply:
Yes that's correct. Technically, webview_flutter is also an extra dependency since it isn't bundled into the framework.

I decided to use flutter_inappwebview for this example because it has some options that are really beneficial for this type of implementation, like disabling scrolling so that when an embedded video is rendered and you scroll on it, the `Html` widget itself scrolls. This has been a problem plaguing webview for a while on Flutter so it can make the UX a little better when the user isn't meant to scroll that webview box.

If you'd like, I can change it to webview_flutter, I'd just have to tweak the example a little to get it inline with its API spec.

Reply:
Yes using a different WebView plugin in the readme than the one we use internally is too confusing in my opinion. People will copy paste this code and end up with 2 webview implementations, which is undesirable. 

If you want to block input going to the webview, couldn't you use AbsorbPointer? Though again this, imho, doesn't really explain how to use flutter_html, which is what this readme should do.

Reply:
We don't want to block all input because the user has to click play on embedded YT videos, so the option of just disabling the scrolling without the need of an extra widget is useful. However I see where you are coming from and I've changed the example to use `webview_flutter`.